dialogue-target/src/main/java/com/palantir/dialogue/Request.java

```diff
@@ -113,22 +117,44 @@ public static Builder builder() {
         return new Builder();
     }
 
+    @NotThreadSafe
     public static final class Builder {
-        private ListMultimap<String, String> headerParams = MultimapBuilder.treeKeys(String.CASE_INSENSITIVE_ORDER)
-                .arrayListValues()
-                .build();
-        private ImmutableListMultimap.Builder<String, String> queryParams = ImmutableListMultimap.builder();
-        private ImmutableMap.Builder<String, String> pathParams = ImmutableMap.builder();
+        private static final ImmutableListMultimap<String, String> EMPTY =
```

Comment:
as @carterkozak suggested in standup today, it would be pretty cool to use a bit set to track whether we have already done the copy-on-write for a given collection

Reply:
Yeah bitset thing would be even nicer, but at the moment I quite like that we're still storing immutable collections internally, so we're really protected from any bugs in the implementation.

Could we land this one as an incremental improvement and consider the bitset thing separately? I'm just worried the bitset approach make it harder to read/understand as we'd need to move away from ImmutableMap.Builder and store mutable things internally...

Reply:
Sure, no need to crank it up to 11 at once.